src/psyclone/psyir/symbols/generic_interface_symbol.py

```diff
@@ -147,9 +147,30 @@ def copy(self):
         '''
         # The constructors for all Symbol-based classes have 'name' as the
         # first positional argument.
-        return type(self)(self.name, self.routines, datatype=self.datatype,
+        return type(self)(self.name, self.routines[:],
```

Comment:
Is the [:] needed?

Reply:
I did it to ensure we get a copy of the `routines` list otherwise the new class might have the same list as the old one.